code/edge/src/main/java/com/adobe/marketing/mobile/EdgeHitProcessor.java

```diff
@@ -151,6 +152,28 @@ public void onComplete() {
 			edgeHit.getRequestId()
 		);
 
+		if (!UrlUtils.isValidUrl(url)) {
+			Log.warning(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Unable to send network request with id (%s) as url is malformed, '%s'.",
+				edgeHit.getRequestId(),
+				url
+			);
+			return true;
+		}
+
+		if (!url.startsWith("https")) {
+			Log.warning(
+				LOG_TAG,
+				LOG_SOURCE,
+				"Unable to send network request with id (%s) as url scheme must be 'https', '%s'.",
+				edgeHit.getRequestId(),
+				url
+			);
+			return true;
+		}
+
```

Comment:
20+ new lines added to this function, how about we add a func for it and here call if (!isValidUrl(url)) return true;